.github/workflows/oauth2.yaml

```diff
@@ -0,0 +1,36 @@
+name: rabbitmq-dotnet-client-oauth2
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build:
+    name: oauth2-uaa
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: true
+      - uses: actions/setup-dotnet@v3
+        with:
+          dotnet-version: 6.x
+      - uses: actions/cache@v3
+        with:
+          path: |
+            ~/.nuget/packages
+            ~/.local/share/NuGet/v3-cache
+          key: ${{ runner.os }}-v0-nuget-${{ hashFiles('**/*.csproj') }}
+          restore-keys: |
+            ${{ runner.os }}-v0-nuget-
+      - run: ${{ github.workspace }}/.ci/oauth2/setup.sh
+      - run: |
+          dotnet run --Name uaa \
+            --ClientId producer \
+            --ClientSecret producer_secret \
+            --Scope '' \
+            --TokenEndpoint 'http://localhost:8080/oauth/token' \
+            --TokenExpiresInSeconds 60 \
```

Comment:
In theory, you should not need to pass this parameter ... This is determined by UAA itself and included in one of the attributes of the token that UAA sends to this client.

Reply:
OK good to know. I'll test it without that. I'm just copying what's in the existing setup scripts.